src/helpers/ConnectorConfigInput.tsx

```diff
@@ -79,21 +79,52 @@ class ConnectorConfigInput {
     return this.specsToRender.concat(childSpecs);
   }
 
-  static traverseSpecsToSetOrder(data: any, parentOrder: number | null = null, ordCounter = 0) {
-    const dataProperties: any = data.properties;
+  static traverseSpecsToSetOrder(data: any, globalCounter = { count: 0 }) {
+    //global counter as an object such that we have a single counter. (pass by refernce concept for objects)
+    if (!globalCounter || typeof globalCounter.count !== 'number') {
+      globalCounter = { count: 0 }; // Ensure initialization
```

Comment:
Just wondering, if this initialization is needed ? Since the function already has a default set

Reply:
Yes removed that. Its working fine. I think earlier I was passing a null value to it from the `setOrderToChildProperties` thats why I had to add that check.